### DISCUSSION TITLE
Midi thru example

### DISCUSSION THREAD

**Discussion by enorrmann** [djipco/webmidi#187] [Q&A] [answered]
Hi ! I'm working on a midi thru example. Is it possible to have 2 devices connected and just forward a message received from the input to the output without using noteon or noteoff ? like forwarding the "raw" midi event to the output
thanks !

**Comment by djipco** [comment, marked as answer]
In [version 3](https://github.com/djipco/webmidi/tree/develop) (soon to go into beta), it is now possible to use the new [`Input.addForwarder()`](https://djipco.github.io/webmidi/api/classes/Input#addForwarder) method to forward messages from an input to an output:

```javascript
const destination = WebMidi.outputs[0];
WebMidi.inputs[0].addForwarder(destination);
```

The code above forwards all messages coming in from input #0 to the destination output. The method can also accept options to filter forwarded messages by type and/or by MIDI channel.

**Reply by enorrmann** [reply]
awesome ! thanks !
